File: db4on/tutorial/db4o-tutorial-chapters/src/com/db4o/f1/chapter1/ArbitraryQuery.cs
using com.db4o.query;

namespace com.db4o.f1.chapter1
{
	public class ArbitraryQuery : Predicate
    {
    	private int[] _points;
    	
    	public ArbitraryQuery(int[] points)
    	{
    		_points=points;
    	}
    
    	public bool match(Pilot pilot)
    	{
        	foreach (int points in _points)
        	{
        		if (pilot.Points == points)
        		{
        			return true;
        		}
        	}
        	return pilot.Name.StartsWith("Rubens");
    	}
    }
}

File: db4on/tutorial/vb/db4o-tutorial-chapters/src/com/db4o/f1/chapter1/ArbitraryQuery.vb
Imports com.db4o.query
Namespace com.db4o.f1.chapter1
	Public Class ArbitraryQuery
	Inherits Predicate
		Private _points As Integer()

		Public Sub New(ByVal points As Integer())
			_points = points
		End Sub 

		Public Function Match(ByVal pilot As Pilot) As Boolean
			For Each points As Integer In _points
				If pilot.Points = points Then
					Return True
				End If
			Next
			Return pilot.Name.StartsWith("Rubens")
		End Function

	End Class
End Namespace
